File: src/python/Trees/InOrderTraversal.py
class TreeNode:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None

def inorder_traversal(root):
    if root:
        inorder_traversal(root.left)  # Traverse left subtree
        print(root.value, end=' ')  # Process the root
        inorder_traversal(root.right)  # Traverse right subtree

if __name__ == '__main__':
    root = TreeNode(1)
    root.left = TreeNode(2)
    root.right = TreeNode(3)
    root.left.left = TreeNode(4)
    root.left.right = TreeNode(5)

    inorder_traversal(root)

File: src/java/Trees/InOrderTraversal.java
import Trees.TreeNode;

public class InOrderTraversal{
 public void inorderTraversal(TreeNode root) {
        if (root == null) {
            return;
        }
        inorderTraversal(root.left);  // Traverse left subtree
        System.out.print(root.val + " ");  // Process the root
        inorderTraversal(root.right);  // Traverse right subtree
    }

    public static void main(String[] args) {
        TreeNode root = new TreeNode(1);
        root.left = new TreeNode(2);
        root.right = new TreeNode(3);
        root.left.left = new TreeNode(4);
        root.left.right = new TreeNode(5);

        InOrderTraversal tree = new InOrderTraversal();
        tree.inorderTraversal(root);
    }
}
